### PR TITLE
[OBSDEF-5979] add apps with run-level for federation to start before iam

### PR DESCRIPTION
**Pull request by bst525**
## Purpose
[OBSDEF-5979](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-5979)
_List the changes for this PR_

## PR checklist
- [x] make test passed
- [x] make build passed
- [x] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed

## Testing
_Paste the output of your helm install/vSphere7 deployment testing here_

### CODE REVIEW

File: federation/templates/federation-app.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+  {{- if .Values.createApplicationResource }}
```
Comment (eddiepavkovic) [inline federation/templates/federation-app.yaml:2]: shouldn't be indented.

File: federation/templates/federation-app.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+apiVersion: app.k8s.io/v1beta1
+kind: Application
+metadata:
+  name: "federation"
```
Comment (eddiepavkovic) [inline federation/templates/federation-app.yaml:6]:
```suggestion
  name: "objectscale-cloud"
```

**Review by eddiepavkovic — COMMENTED** [review]
FYI.  the run-levels are only used in vSphere7 deployments where these apps are kicked off after we deploy the plugin.  
Can you please indicate what vSphere7 testing was done to ensure the feds start before iams

**Review by eddiepavkovic — CHANGES_REQUESTED** [review]
we need to discuss with iam/fed team if this is the correct approach.  run-levels are only tracked by `install-controller` which is used during initial vsphere7 deployment.  and used by graphql to install objectstores, supportassist and licensing.

**Review by vgupta-mickey — COMMENTED** [review]
I thought I are suppose to have a meeting on this to understand the sub-charts vs run-level?
run-level make sense when the install controller is installing the federation service as a main chart

**Pull request closed without merging**